### PR TITLE
hide oz classroom auto clans

### PR DESCRIPTION
**Pull request by smallst**
image
fix ENG-1120

## Summary by CodeRabbit

- **New Features**
	- Enhanced classroom data management functionality in the Clan Selector component by integrating Vuex for state management.
	- Added the ability to fetch and display classroom information associated with the current user.
	- Implemented refined filtering of clans based on classroom slugs.

- **Bug Fixes**
	- Improved error handling during the fetching of classroom data.

### CODE REVIEW

File: app/views/ai-league/ClanSelectorTeachers.vue [hunk 1]
```diff
@@ -33,10 +34,24 @@ export default {
...
-
+  data () {
+    return {
+      ozClassrooms: []
+    }
+  },
   computed: {
     clansSanitized () {
-      return this.clans.filter(v => v !== undefined)
+      return this.clans.filter(v => v !== undefined).filter(v => !this.ozClassroomsSlugs.includes(v.slug))
```
Comment (adamkecskes) [inline app/views/ai-league/ClanSelectorTeachers.vue:44]: Is it possible that there are `undefined` values among `clans`? why?

**Review by adamkecskes — APPROVED** [review]
The code LGTM - didn't try

File: app/views/ai-league/ClanSelectorTeachers.vue (hunk 1 above)
Comment (smallst) [reply app/views/ai-league/ClanSelectorTeachers.vue:44]: it has. i don't know why. my testing account has the `undefined` in clans array so the original code is reasonable 😂

**Review by mrfinch — COMMENTED** [review]
wouldn't it be easier to just remove classes whose id we don't have in coco instead of fetching oz classrooms? 😅

**Review by mrfinch — APPROVED** [review]

**Pull request merged**